core/error.rs

```diff
@@ -207,6 +207,85 @@ impl JsError {
     Self::inner_from_v8_exception(scope, exception, Default::default())
   }
 
+  pub fn from_v8_message<'a>(
+    scope: &'a mut v8::HandleScope,
+    msg: v8::Local<'a, v8::Message>,
+  ) -> Self {
+    // Create a new HandleScope because we're creating a lot of new local
+    // handles below.
+    let scope = &mut v8::HandleScope::new(scope);
+
+    let exception_message = msg.get(scope).to_rust_string_lossy(scope);
+    let state_rc = JsRuntime::state(scope);
+
+    // Convert them into Vec<JsStackFrame>
+    let mut frames: Vec<JsStackFrame> = vec![];
+
+    let mut source_line = None;
+    let mut source_line_frame_index = None;
+    {
+      let state = &mut *state_rc.borrow_mut();
+
+      let script_resource_name = msg
+        .get_script_resource_name(scope)
+        .and_then(|v| v8::Local::<v8::String>::try_from(v).ok())
+        .map(|v| v.to_rust_string_lossy(scope));
+      let line_number: Option<i64> =
+        msg.get_line_number(scope).and_then(|v| v.try_into().ok());
+      let column_number: Option<i64> = msg.get_start_column().try_into().ok();
+      if let (Some(f), Some(l), Some(c)) =
+        (script_resource_name, line_number, column_number)
+      {
+        // V8's column numbers are 0-based, we want 1-based.
+        let c = c + 1;
+        if let Some(source_map_getter) = &state.source_map_getter {
+          let (f, l, c) = apply_source_map(
```

Comment:
Why don't we use 0-based line and column numbers? This function immediately does:

```
// Lookup expects 0-based line and column numbers, but ours are 1-based.
let line_number = line_number - 1;
let column_number = column_number - 1;
```

Reply:
V8 uses different offsets than TSC/SWC

Reply:
tsc is 0-indexed and so does the lsp. In swc on our diagnostics we have a "display position" that is 1-indexed. We try to only use 1-indexed for displaying purposes only. I don't think we shouldn't be mixing unit here.

Reply:
Can we do this refactor in a follow up then? It's not immediately related to the problem this PR fixes.